src/gateway.cpp

```diff
@@ -83,9 +83,11 @@ void Gateway::routeToInternet() {
   auto message = txQueue->poll();
   if (message == nullptr)
     return;
-
   auto l2Data = &message->data.layer2;
-  printLayer2Data(l2Data);  // TODO route to Internet instead of print...
+  auto ipv4 = (ipv4_headers_t*)l2Data->payload;
+  void* payload = malloc(l2Data->length);
+  memcpy(payload, l2Data->payload, l2Data->length);
```

Comment:
Why is it needed to copy the payload to a new memory buffer?

Reply:
Because we think this memory address is going to be emptied. That way, we move on as a copy to be safe.

Reply:
It is not going to be freed before transmission. This copy is not necessary.